compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -320,62 +327,105 @@ private class ExtractAPICollector(using Context) extends ThunkHolder {
     }
   }
 
-  def apiDefinitions(defs: List[Symbol]): List[api.ClassDefinition] = {
-    defs.sorted(classFirstSort).map(apiDefinition)
-  }
+  def apiDefinitions(defs: List[Symbol]): List[api.ClassDefinition] =
+    defs.sorted(classFirstSort).map(apiDefinition(_, inlineOrigin = NoSymbol))
 
-  def apiDefinition(sym: Symbol): api.ClassDefinition = {
+  /** `inlineOrigin` denotes an optional inline method that we are
+   *  currently hashing the body of. If it exists, include extra information
+   *  that is missing after erasure
+   */
+  def apiDefinition(sym: Symbol, inlineOrigin: Symbol): api.ClassDefinition = {
     if (sym.isClass) {
       apiClass(sym.asClass)
     } else if (sym.isType) {
       apiTypeMember(sym.asType)
     } else if (sym.is(Mutable, butNot = Accessor)) {
       api.Var.of(sym.name.toString, apiAccess(sym), apiModifiers(sym),
-        apiAnnotations(sym).toArray, apiType(sym.info))
+        apiAnnotations(sym, inlineOrigin).toArray, apiType(sym.info))
     } else if (sym.isStableMember && !sym.isRealMethod) {
       api.Val.of(sym.name.toString, apiAccess(sym), apiModifiers(sym),
-        apiAnnotations(sym).toArray, apiType(sym.info))
+        apiAnnotations(sym, inlineOrigin).toArray, apiType(sym.info))
     } else {
-      apiDef(sym.asTerm)
+      apiDef(sym.asTerm, inlineOrigin)
     }
   }
 
-  def apiDef(sym: TermSymbol): api.Def = {
+  /** `inlineOrigin` denotes an optional inline method that we are
+   *  currently hashing the body of. If it exists, include extra information
+   *  that is missing after erasure
+   */
+  def apiDef(sym: TermSymbol, inlineOrigin: Symbol): api.Def = {
+
+    val inlineExtras = new mutable.ListBuffer[Int => Int]
```

Comment:
Instead of storing functions, could inlineExtras be an Int representing an hashcode, initialized to 0 (or some initial hash value but I don't think it matters either way) and updated whenever we see an inline param?